test/caption-stream.test.js

```diff
@@ -514,6 +701,197 @@ QUnit.skip('applies preamble address codes', function() {
   QUnit.ok(false, 'not implemented');
 });
 
+QUnit.skip('applies mid-row colors', function() {
+  QUnit.ok(false, 'not implemented');
+});
+
+QUnit.test('applies mid-row underline', function() {
+  var captions = [];
+  cea608Stream.on('data', function(caption) {
+    captions.push(caption);
+  });
+
+  var packets = [
+    { ccData: 0x1425, type: 0 },
+    { ccData: characters('no'), type: 0 },
+    { ccData: 0x1121, type: 0 },
+    { ccData: characters('ye'), type: 0 },
+    { ccData: characters('s.'), type: 0 },
+    { ccData: 0x142d, type: 0 }
+  ];
+
+  packets.forEach(cea608Stream.push, cea608Stream);
+  QUnit.equal(captions[0].text, 'no <u>yes.</u>', 'properly closed by CR');
+});
+
+QUnit.test('applies mid-row italics', function() {
+  var captions = [];
+  cea608Stream.on('data', function(caption) {
+    captions.push(caption);
+  });
+
+  var packets = [
+    { ccData: 0x1425, type: 0 },
+    { ccData: characters('no'), type: 0 },
+    { ccData: 0x112e, type: 0 },
+    { ccData: characters('ye'), type: 0 },
+    { ccData: characters('s.'), type: 0 },
+    { ccData: 0x142d, type: 0 }
+  ];
+
+  packets.forEach(cea608Stream.push, cea608Stream);
+  QUnit.equal(captions[0].text, 'no <i>yes.</i>', 'properly closed by CR');
+});
+
+QUnit.test('applies mid-row italics underline', function() {
+  var captions = [];
+  cea608Stream.on('data', function(caption) {
+    captions.push(caption);
+  });
+
+  var packets = [
+    // RCL, resume caption loading
+    { ccData: 0x1425, type: 0 },
+    { ccData: characters('no'), type: 0 },
+    { ccData: 0x112f, type: 0 },
+    { ccData: characters('ye'), type: 0 },
+    { ccData: characters('s.'), type: 0 },
+    { ccData: 0x142d, type: 0 }
+  ];
+
+  packets.forEach(cea608Stream.push, cea608Stream);
+  QUnit.equal(captions[0].text, 'no <i><u>yes.</u></i>', 'properly closed by CR');
+});
+
+QUnit.test('applies PAC color underline', function() {
```

Comment:
Good point about the ambiguous name. I say "PAC color underline" because coloured underline PACs are processed, and underline is added to the text. However, as you point out, colour is not supported, so that information is dropped.

So if you were to change the PAC from `0x1461` (row 15 white underline) to `0x1463` (row 15 green underline), you will still get out `'<u>yes.</u>'`. If colour support is added, more tests would be needed to differentiate between the various supported colours.